upcloud/gateway.go

```diff
@@ -30,14 +41,50 @@ const (
 	GatewayOperationalStateDeleteLinkNetwork GatewayOperationalState = "delete-link-network"
 	GatewayOperationalStateDeleteService     GatewayOperationalState = "delete-service"
 
-	// GatewayFeatureNAT is the network address translation (NAT) feature of the network gateway
+	GatewayTunnelOperationalStateUninitialized GatewayTunnelOperationalState = "uninitialized"
+	GatewayTunnelOperationalStateCreated       GatewayTunnelOperationalState = "created"
+	GatewayTunnelOperationalStateConnecting    GatewayTunnelOperationalState = "connecting"
+	GatewayTunnelOperationalStateEstabilished  GatewayTunnelOperationalState = "established"
+	GatewayTunnelOperationalStateRekeying      GatewayTunnelOperationalState = "rekeying"
+	GatewayTunnelOperationalStateRekeyed       GatewayTunnelOperationalState = "rekeyed"
+	GatewayTunnelOperationalStateDeleting      GatewayTunnelOperationalState = "deleting"
+	GatewayTunnelOperationalStateDestroying    GatewayTunnelOperationalState = "destroying"
+	GatewayTunnelOperationalStateUnknown       GatewayTunnelOperationalState = "unknown"
+
+	// GatewayFeatureNAT is a Network Address Translation (NAT) service that offers a way for cloud servers in SDN private networks to connect to the Internet through the public IP assigned to the network gateway service
 	GatewayFeatureNAT GatewayFeature = "nat"
+	// GatewayFeatureVPN is a Virtual Private Network (VPN) service used to establish an encrypted network connection when using public networks
+	GatewayFeatureVPN GatewayFeature = "vpn"
+
+	GatewayConnectionTypeIPSec GatewayConnectionType = "ipsec"
+
+	GatewayRouteTypeStatic GatewayRouteType = "static"
+
+	GatewayTunnelIPSecAuthTypePSK GatewayIPSecAuthType = "psk"
+
+	GatewayIPSecAlgorithm_aes128gcm16  GatewayIPSecAlgorithm = "aes128gcm16"
```

Comment:
Added an underscore in those constants, as I believe it's easier to parse with eyes when the actual algorithm type is separated a bit more from the rest.